### ISSUE TITLE
False negative SC2086 with default assignments ("${foo:=bar}")

### ISSUE THREAD

**Issue by girst** [koalaman/shellcheck#1981]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2086
- My shellcheck version (`shellcheck --version` or "online"): 0.7.1, online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
: "${foo:=oneword}"
echo $foo

```

#### Here's what shellcheck currently says:

nothing

#### Here's what I wanted or expected to see:

*tl;dr:* For default assignments, shellcheck should not assume that a variable contains no spaces.

*long:* For unsplittable strings, shellcheck does not issue a warning. The above snippet uses a [default assignment](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02), meaning if the variable `$foo` gets passed in through the environment, it is not overwritten with `oneword`. Then, splitting is again a problem.

Both default assignments (`:=` and `=`) are affected.

```
~ % ./example
oneword
~ % foo="two     words" ./example
two words
~ % # notice that the multiple spaces have been eaten by the word splitting
